tests/functional/jobs/test_func_load_job_series.py

```diff
@@ -0,0 +1,199 @@
+from datetime import datetime
+
+import pandas
+import pytest
+
+from sarc.config import MTL, config
+from sarc.jobs import SlurmJob
+from sarc.jobs.series import load_job_series
+
+parameters = {
+    "no_cluster": {},
+    "cluster_str": {"cluster": "patate"},
+    "job_state": {"job_state": "COMPLETED"},
+    "one_job": {"job_id": 10},
+    "one_job_wrong_cluster": {"job_id": 10, "cluster": "patate"},
+    "many_jobs": {"job_id": [8, 9]},
+    "no_jobs": {"job_id": []},
+    "start_only": {"start": datetime(2023, 2, 19, tzinfo=MTL)},
+    "end_only": {"end": datetime(2023, 2, 16, tzinfo=MTL)},
+    "start_str_only": {"start": "2023-02-19"},
+    "end_str_only": {"end": "2023-02-16"},
+    "start_and_end": {
+        "start": datetime(2023, 2, 15, tzinfo=MTL),
+        "end": datetime(2023, 2, 18, tzinfo=MTL),
+    },
+    "user": {"user": "beaubonhomme"},
+    "resubmitted": {"job_id": 1_000_000},
+}
+
+
+few_parameters = {key: parameters[key] for key in ("no_cluster", "start_and_end")}
+param_start_end = {"start_and_end": parameters["start_and_end"]}
+
+ALL_COLUMNS = [
+    "CLEAR_SCHEDULING",
+    "STARTED_ON_BACKFILL",
+    "STARTED_ON_SCHEDULE",
+    "STARTED_ON_SUBMIT",
+    "account",
+    "allocated",
+    "array_job_id",
+    "cluster_name",
+    "constraints",
+    "cpu",
+    "cpu_utilization",
+    "duration",
+    "elapsed_time",
+    "end",
+    "end_time",
+    "exit_code",
+    "gpu_allocated",
+    "gpu_memory",
+    "gpu_power",
+    "gpu_requested",
+    "gpu_utilization",
+    "group",
+    "id",
+    "job_id",
+    "job_state",
+    "mem",
+    "name",
+    "nodes",
+    "partition",
+    "priority",
+    "qos",
+    "requested",
+    "signal",
+    "start",
+    "start_time",
+    "stored_statistics",
+    "submit",
+    "submit_time",
+    "system_memory",
+    "task_id",
+    "time_limit",
+    "user",
+    "work_dir",
+]
+
+
+# For file regression tests, we will save data frame into a CSV.
+# We won't include job id because it changes from a call to another.
```

Comment:
Pourtant les jobs id et les datetimes sont fix dans les autres tests comme ceux de `count_jobs` ajouté à ce PR non?

Reply:
Pour les IDs, ils ne semblent pas fixes, car ils sont exclus du JSON testé dans les file regressions. Exemple ici: https://github.com/mila-iqia/SARC/blob/master/tests/functional/jobs/test_func_get_jobs.py#L36

Pour les datetimes, j'ai aussi constaté qu'ils pouvaient légèrement changer dans certaines rows (au niveau des microsecondes).

Reply:
> Pour les IDs, ils ne semblent pas fixes, car ils sont exclus du JSON testé dans les file regressions. Exemple ici: https://github.com/mila-iqia/SARC/blob/master/tests/functional/jobs/test_func_get_jobs.py#L36

Ah ok!

Reply:
> Pour les datetimes, j'ai aussi constaté qu'ils pouvaient légèrement changer dans certaines rows (au niveau des microsecondes).

On pourrait ajuster les options de print pour le datetime pour éviter les microsecondes. C'est quand même étrange que ça varie comme ça.

Reply:
J'ai ajouté un test pour mettre en évidence les variations: https://github.com/mila-iqia/SARC/pull/89/commits/e0fa9122a47e33c7cce1c024547daee7253b0cd4

La raison de ces variations est ici: https://github.com/mila-iqia/SARC/blob/sarc-229/sarc/jobs/series.py#L275

Reply:
Je crois qu'il serait correct de _mock_ `datetime.now` pour éviter cette variation.

Par contre, ce n'est pas très logique que le end time soit basé sur le temps courant, car le scrapping pourrait être vieux, et aussi ça donne un end time qui dépend de l'ordre dans lequel les jobs sont query. Ça m'étonne qu'on n'aie pas penser à logger un timestamp pour le scrapping. Si on en avait un, on pourrait s'en servir pour le end_time dans le code ici.

Je propose en attendant qu'on crée un timestamp au début du for-loop et qu'on s'en serve pour toutes les jobs. Ça vaudrait tout de même la peine de le _mock_ pour les tests.

Reply:
J'ai créé un ticket pour pas qu'on perde de vue cette modification: https://mila-iqia.atlassian.net/browse/SARC-247

Reply:
Fait ! https://github.com/mila-iqia/SARC/pull/89/commits/7a357b05bd8117e551f4f023e4fd1723b672dc70

Reply:
Merci! Je n'ai pas vérifié c'est quoi les `start_time` et autre, mais `2017-05-21` c'est possiblement bien avant tous les datetimes. Ça serait bien que ça soit au moins après les timestamps des jobs. 2023-11-22 c'est probablement une bonne date.

Reply:
En effet ^^, corrigé: https://github.com/mila-iqia/SARC/pull/89/commits/43b4a338e4b7afbd926a06d59aa41e5782e32b37

Reply:
J'avais oublié un mock. Nouveau commit: https://github.com/mila-iqia/SARC/pull/89/commits/8238bff9cef15e0927b2af10790903d882daa6ea